### PR TITLE
Zenodo Upload and Mkdocs

### PR DESCRIPTION
**Pull request by wd15**
**Implements Zenodo upload and the bones of mkdocs documentation**

 - Tests need to be run with the `ZENODO_SANDBOX_API_TOKEN` otherwise a failure will occur. I think it's better to test with an explicit upload rather than mocking. That's why there is a sandbox after all. Maybe in future this could be toggled for remote branches, but I don't have time to implement this now.
   - Go to https://sandbox.zenodo.org/account/settings/applications/ to generate a token
 - Forked PRs will fail a test currently, but not worried about that.
 - Added `ZENODO_SANDBOX_API_TOKEN` secret to repo secrets
 - Implemented workflow to deploy mkdocs, but untested as need to be merged to main. Runs on push currently (should it run on merge?). I want it to run everytime main is changed in anyway.
 - mkdocs docs currently residing at https://pages.nist.gov/pfhub-cli/
 - added some deployment and zenodo_client documentation
 - To build and view the docs locally use `mkdocs serve` in the base of the repo
 - To upload a record to zenodo use `pfhub upload pfhub/test_data/pfhub.yaml --sandbox`

### CODE REVIEW

**Comment by guyer** [comment]
> Go to https://sandbox.zenodo.org/account/settings/applications/ to generate a token

I cannot log in with ORCID. This appears to be because it tries to use sandbox.orcid.org for authentication, which is all kinds of (purposely) broken. Very confusing.

GitHub login works.

**Comment by guyer** [comment]
What scopes?

**Comment by guyer** [comment]
I guess you've already set `ZENODO_SANDBOX_API_TOKEN`?

**Comment by guyer** [comment]
Link at https://pages.nist.gov/pfhub-cli/#license gives 404.

**Comment by guyer** [comment]
`pfhub test` generates `WARNING: Failed to generate report: No data to report.` with or without `ZENODO_SANDBOX_API_TOKEN` environment variable.

File: docs/api.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# API
+
+
+::: pfhub.main
+    options:
+      members:
+        - efficiency_plot
+        - levelset_plot
```
Comment (guyer) [inline docs/api.md:8]: doctest lines are being interpreted as a quoted block

**Review by guyer — APPROVED** [review]

**Comment by wd15** [comment]
> What scopes?

I'm not sure what you're referring to .

**Comment by guyer** [comment]
> > What scopes?
> 
> I'm not sure what you're referring to .

When you make a Zenodo token, the first thing it asks you is what scopes the token should have

**Comment by wd15** [comment]
In response to @guyer comments:

 - Added much better [guide to setting up Zenodo](https://pages.nist.gov/pfhub-cli/DEVELOPMENT/#setting-up-zenodo).
 - Fixed [license link](https://pages.nist.gov/pfhub-cli/#license).
 - Unfortunately, [mkdocstrings](https://mkdocstrings.github.io) requires back ticks around the test block on doctests. Doing that for now. At least it [renders correctly now](https://pages.nist.gov/pfhub-cli/api/#pfhub.main.levelset_plot). Better solution would be to amend mkdocstrings to work correctly.
 - Made `pfhub test` not use coverage so output is cleaner.

**Pull request merged**